### PR TITLE
Handle customer additionalData

### PR DESCRIPTION
**Pull request by ekohl**
This does snake_to_camel and camel_to_snake in the appropriate places.
It also makes it possible to not send additional data which should
prevent the values from being wiped.

### CODE REVIEW

**Comment by ekohl** [comment]
This includes https://github.com/AntagonistHQ/openprovider.py/pull/62 and still needs work. By using random fixtures, the tests fail with betamax cassettes which include the old data.

File: tests/test_domain.py [hunk 1]
```diff
@@ -1,72 +1,181 @@
...
+
+
+@betamaxed
+def test_domain_search_empty(api):
+    """Searching for example.com should return nothing."""
+    result = api.domains.search_domain_request(domain_name_pattern="example.com")
+    assert result == []
+
+
+def test_domain_order(api, domainname, nameservers):
```
Comment (blubber) [inline tests/test_domain.py]: Misschien markeren als integratietest?

**Comment by blubber** [comment]
Tests fail

**Comment by ekohl** [comment]
Maybe we could use https://pypi.python.org/pypi/fake-factory to generate names instead.

**Pull request merged**